core/schema/modulesource.go

```diff
@@ -1378,7 +1378,9 @@ func (s *moduleSchema) collectCallerLocalDeps(
 				modCfg.Name = src.WithName
 			}
 			if src.WithSDK != "" {
-				modCfg.SDK = src.WithSDK
+				modCfg.SDK = &modules.SDK{
+					Source: src.WithSDK,
+				}
```

Comment:
`WithSDK` would need changing as well - the method should be updated to additionally take `pin` (similar to how `moduleSource` takes it as well), and maybe even `env`? Happy to defer that for now, but IMO the `WithSDK` field should change to be the new `SDKConfig` type instead of a `string`.